File: PlayerUI/Flutter/objc/FlutterPlayer/BCOVVideoPlayerFactory.m
//
//  BCOVVideoPlayerFactory.m
//  FlutterPlayer
//
//  Copyright © 2024 Brightcove, Inc. All rights reserved.
//

#import "BCOVVideoPlayer.h"

#import "BCOVVideoPlayerFactory.h"


@implementation BCOVVideoPlayerFactory

#pragma mark - FlutterPlatformViewFactory

- (NSObject<FlutterPlatformView> *)createWithFrame:(CGRect)frame
                                    viewIdentifier:(int64_t)viewId
                                         arguments:(id _Nullable)args
{
    return [[BCOVVideoPlayer alloc] initWithFrame:frame
                                   viewIdentifier:viewId
                                        arguments:args];
}

- (NSObject<FlutterMessageCodec> *)createArgsCodec
{
    return FlutterStandardMessageCodec.sharedInstance;
}

@end

File: PlayerUI/Flutter/FlutterPlayer/BCOVVideoPlayerFactory.swift
//
//  BCOVVideoPlayerFactory.swift
//  FlutterPlayer
//
//  Copyright © 2024 Brightcove, Inc. All rights reserved.
//

import UIKit
import Flutter
import BrightcovePlayerSDK


final class BCOVVideoPlayerFactory: NSObject, FlutterPlatformViewFactory {

    func create(withFrame frame: CGRect,
                viewIdentifier viewId: Int64,
                arguments args: Any?) -> FlutterPlatformView {
        return BCOVVideoPlayer(frame: frame,
                               viewId: viewId,
                               args: args)
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}
